catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTablePartitionOperations.java

```diff
@@ -0,0 +1,341 @@
+    checkArgument(exceptionConverter != null, "exceptionConverter is null");
+    checkArgument(typeConverter != null, "typeConverter is null");
+    this.exceptionConverter = exceptionConverter;
+    this.typeConverter = typeConverter;
+  }
+
+  @Override
+  public String[] listPartitionNames() {
+    try (Connection connection = getConnection(databaseName)) {
+      String showPartitionsSql = String.format("SHOW PARTITIONS FROM `%s`", tableName);
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        ImmutableList.Builder<String> partitionNames = ImmutableList.builder();
+        while (result.next()) {
+          partitionNames.add(result.getString("PartitionName"));
+        }
+        return partitionNames.build().stream().toArray(String[]::new);
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  public Partition[] listPartitions() {
+    try (Connection connection = getConnection(databaseName)) {
+      Transform partitionInfo = getPartitionInfo(connection);
+      Map<String, Type> columnTypes = getColumnType(connection);
+      String showPartitionsSql = String.format("SHOW PARTITIONS FROM `%s`", tableName);
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        ImmutableList.Builder<Partition> partitions = ImmutableList.builder();
+        while (result.next()) {
+          partitions.add(fromDorisPartition(result, partitionInfo, columnTypes));
+        }
+        return partitions.build().stream().toArray(Partition[]::new);
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+  }
+
+  @Override
+  public Partition getPartition(String partitionName) throws NoSuchPartitionException {
+    try (Connection connection = getConnection(databaseName)) {
+      Transform partitionInfo = getPartitionInfo(connection);
+      Map<String, Type> columnTypes = getColumnType(connection);
+      String showPartitionsSql =
+          String.format(
+              "SHOW PARTITIONS FROM `%s` WHERE PartitionName = \"%s\"", tableName, partitionName);
+      try (Statement statement = connection.createStatement();
+          ResultSet result = statement.executeQuery(showPartitionsSql)) {
+        while (result.next()) {
+          return fromDorisPartition(result, partitionInfo, columnTypes);
+        }
+      }
+    } catch (SQLException e) {
+      throw exceptionConverter.toGravitinoException(e);
+    }
+    throw new NoSuchPartitionException("Partition %s does not exist", partitionName);
+  }
+
+  @Override
+  public Partition addPartition(Partition partition) throws PartitionAlreadyExistsException {
+    try (Connection connection = getConnection(databaseName)) {
+      Transform partitionInfo = getPartitionInfo(connection);
```

Comment:
Can we save the table object when `createJdbcTablePartitionOperations`? so that we don't need to call `getPartitionInfo` again in partition operations